package/network/config/ltq-vdsl-app/files/10_atm_ptm.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/sh
```

Comment:
> Since it is a rare case that the line mode changes, I would prefer to drop the module unload/restart for now...

Doing so, a cpe won't be able to come online again if the dsl/tc_layer is changed by the network operator. This might be acceptable for private use, but for devices that are installed on places where no person is on site for the overwhelming time, this could be a real problem. So I would prefer the dirtier but finally safest way with the reboot, as this shouldn't happen to often anyway. :-)

> For instance, bringing up the atm ethernet bridge (via init script) isn't necessary if ETM/PTM is used.

This wouldn't work anyway. The current implementation (starting br2684 via init script) only works, because dsl_control (where the atm driver is loaded) is started before the br2684 init script.

When the atm driver is loaded by the notify script, the br2684 init script would be called a long time ago and the br2684 won't be setup correctly.

So with my approach you have to setup the br2684 in the notify script after loading the atm driver. But if you are right, and no one else then lantiq is using this br2684 init script, it can be dropped at all.

> But I don't have yet an idea how to integrate it nicely without breaking danube/ar9. Do you know by any chance how it is done that the ltq-atm module is autoloaded on danube/ar9 but not on vr9? I fail to see it.

As the ATM/PTM auto-mode should be also useable for ADSL only devices like danube/ar9/... what about just do it also for them?